packages/web3-core-subscriptions/src/index.js

```diff
@@ -59,7 +59,7 @@ Subscriptions.prototype.buildCall = function() {
         }
 
         var subscription = new Subscription({
-            subscription: _this.subscriptions[arguments[0]],
+            subscription: _this.subscriptions[arguments[0]] || {}, // Subscript might not exist
```

Comment:
@cgewecke Great finding! This is an undocumented and questionable feature and we should probably document it in our documentation if it actually does make sense. This because we have already implemented all subscriptions for the ``shh`` and ``eth`` module and it isn't possible to start for example a ``parity`` subscription with the ``web3.eth.subscribe`` method because the property ``type`` of the ``Subscriptions`` class is hardcoded to ``eth`` which should be ``parity`` in the case someone wants to use not implemented ``parity`` specific subscriptions. Because no one was using this un-documented feature until now (because of the syntax error would we definitely have an open issue) can we think about to extend the capabilities of the ``web3.eth.subscribe`` method to support fully custom-defined subscriptions. The possible existing options object we already have in the function signature of the ``web3.eth.subscribe`` method would probably give us the opportunity to do it in a non-breaking way.

Reply:
Ok great - that's a good idea. The parity subscriptions you're talking about are [these](https://wiki.parity.io/JSONRPC-parity_pubsub-module)?

It's like 'subscribing'  over websockets to the standard RPC methods (ex:`eth_getBalance`) to get a single notification?

Reply:
Yes, they do offer some "normal" JSON-RPC method calls as subscriptions for WebSocket connections. It's definitely better to use such a subscription if existing instead of polling :)


Btw. and off-topic: I've created a new subscription type in the 2.0 MVP to have polling subscriptions ([PollingSubscription](https://github.com/ethereum/web3.js/blob/architecture/expose-only-one-package/internal/core/src/json-rpc/subscriptions/polling/PollingSubscription.ts))